client/wfm/src/model/WorkOrder.ts

```diff
@@ -69,4 +69,17 @@ export interface WorkOrder {
    * Timestamp of this WorkOrder's last update
    */
   updated: number;
+
+  /**
+   * List of timestamps of when this WorkOrder last reached a given {@link STATUS}
+   *
+   * @example
+   * {
+   *   'New': 1506010024468,
+   *   'Complete': 1506010024468
+   * }
+   */
+  statusHistory?: {
```

Comment:
[Question] Do we really need that for this bug?
How about relaying on `status='FINISHED' and updated< moment.daysFromNow(2)`

IMHO after finished workorder is being accepted it should be in read only mode in portal.

Reply:
I actually wanted this to be an array, since we might end up having workorders unassigned and reassigned to other users, they might hit the same status more than once in their lifetime and tracking all of that can lead to more detailed filtering and reporting. But the filter, while doable, could have some performance issues...

I know we don't expect workorders to be updated after they're complete, but there's always stuff like someone in the portal being able to update the workorder name, or maybe we add comments in the future that could also be added after it is complete, so relying on a separate timestamp is more future-proof too.